src/opendr/perception/object_detection_2d/nanodet/nanodet_learner.py

```diff
@@ -333,11 +360,147 @@ def reset(self):
+    def _load_onnx(self, onnx_path, verbose=True):
+        if verbose:
+            print("Loading ONNX runtime inference session from {}".format(onnx_path))
+            self.logger.log("Loading ONNX runtime inference session from {}".format(onnx_path))
+
+        self.ort_session = ort.InferenceSession(onnx_path)
+
+    def _save_jit(self, jit_path, img=None, verbose=True):
+        if self.ort_session:
+            print("Warning: An onnx model has already initialized, inference will run in onnx mode by default!!!")
+        if not self.predictor:
+            self.predictor = Predictor(self.cfg, self.model, device=self.device)
+
+        os.makedirs(jit_path, exist_ok=True)
+
+        if not self.dummy_input:
+            assert img, \
+                "When use optimize or _save_jit is called for the first time, it must have and opendr Image input"
+            if not isinstance(img, Image):
+                img = Image(img)
+            img = img.opencv()
+            if not self.dummy_input:
+                self.dummy_input = self.predictor.preprocessing(img)
+
+        with torch.no_grad():
+            export_path = os.path.join(jit_path, "nanodet_{}.pth".format(self.cfg.check_point_name))
+            self.predictor.trace_model(self.dummy_input)
+            model_traced = torch.jit.script(self.predictor)
+
+            metadata = {"model_paths": ["nanodet_{}.pth".format(self.cfg.check_point_name)], "framework": "pytorch",
+                        "format": "pth", "has_data": False, "inference_params": {}, "optimized": True,
+                        "optimizer_info": {}, "classes": self.classes}
+            model_traced.save(export_path)
+
+            with open(os.path.join(jit_path, "nanodet_{}.json".format(self.cfg.check_point_name)),
+                      'w', encoding='utf-8') as f:
+                json.dump(metadata, f, ensure_ascii=False, indent=4)
+
+            if verbose:
+                print("Finished export to TorchScript")
+                self.logger.log("Finished export to TorchScript")
+
+    def _load_jit(self, jit_path, verbose=True):
+        if verbose:
+            print("Loading Jit model from {}".format(jit_path))
+            self.logger.log("Loading Jit model from {}".format(jit_path))
+
+        self.jit_model = torch.jit.load(jit_path, map_location=self.device)
+
+    def optimize(self, export_path, initial_img=None, verbose=True, optimization="jit"):
+        """
+        Method for optimizing the model with onnx or jit.
+        :param export_path: the path to the folder that the model is or will be after optimization
+        :type export_path: str
+        :param initial_img: if optimize is called for the first time is needed a dummy input of opendr Image
+        :type initial_img: Image
+        :param verbose: if set to True, additional information is printed to STDOUT and logger txt output,
+        defaults to True
```

Comment:
Verbose usually refers to print output and it seems a bit excessive to include logger text output. I would suggest splitting it into an extra argument just for logger text output. It also defaults to True, which would be ok for some methods just for printing, but right now it also produces text file clutter, that might not be needed. In general, i think text logging should be opt-in not opt-out for the user and separate from printing.